pageserver/client_grpc/src/split.rs

```diff
@@ -35,8 +32,12 @@ impl GetPageSplitter {
             return Some(ShardIndex::unsharded());
         }
 
-        // Find the base shard index for the first page, and compare with the rest.
-        let key = rel_block_to_key(req.rel, *req.block_numbers.first().expect("no pages"));
+        // Find the base shard index for the first page, and compare with the rest. If there are
+        // no pages, just return the first shard.
+        let Some(&first_page) = req.block_numbers.first() else {
+            return Some(ShardIndex::new(ShardNumber(0), count));
```

Comment:
Why do we allow the empty request?

Reply:
> as we've already checked this upstream

I thought that'd be the case. I'd just assert it (we have similar assertion sprinkled in). A comment would also do.